src/analyses/vla.ml

```diff
@@ -0,0 +1,51 @@
+(** An analysis to detect if an invocation is in the scope of a variably modified variable. *)
+
+open Prelude.Ana
+open Analyses
+
+module Spec =
+struct
+  include Analyses.DefaultSpec
+
+  let name () = "vla"
+  module D = BoolDomain.MayBool
+  module C = Lattice.Unit
+
+  (* transfer functions *)
+  let assign ctx (lval:lval) (rval:exp) : D.t =
+    ctx.local
+
+  let branch ctx (exp:exp) (tv:bool) : D.t =
+    ctx.local
+
+  let body ctx (f:fundec) : D.t =
+    ctx.local
+
+  let return ctx (exp:exp option) (f:fundec) : D.t =
+    ctx.local
+
+  let enter ctx (lval: lval option) (f:fundec) (args:exp list) : (D.t * D.t) list =
+    [ctx.local, false]
+
+  let combine ctx ?(longjmpthrough = false) (lval:lval option) fexp (f:fundec) (args:exp list) fc (au:D.t) (f_ask: Queries.ask) : D.t =
+    ctx.local
+
+  let special ctx (lval: lval option) (f:varinfo) (arglist:exp list) : D.t =
+    ctx.local
+
+  let vdecl ctx (v:varinfo) : D.t = true
```

Comment:
This should actually check if it's a VLA or just a declaration, instead of implicitly relying on the fact that right now we choose to keep `Cabs2cil.alwaysGenerateVarDecl` as false.

Actually, isn't this analysis just a check whether any of the current function's locals is a VLA? I think that could be checked directly via CIL instead of a new analysis.